File: 0max-consecutive-ones-iii/0max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        n = len(nums)
        left = 0
        zeroes_count = 0
        max_consecutive_ones = 0
        
        for right in range(n):
            if nums[right] == 0:
                zeroes_count += 1
                
            while zeroes_count > k:
                if nums[left] == 0:
                    zeroes_count -= 1                
                left += 1
            
            max_consecutive_ones = max(max_consecutive_ones, right - left + 1)
        
        return max_consecutive_ones
            

File: 0max-consecutive-ones-iii/0max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int left = 0;
        int curr = 0;
        int longestOnes = 0;
        
        for(int right = 0; right < nums.length; right++){
            curr += nums[right] == 0 ? 1 : 0;
            
            while(curr > k){
                curr -= nums[left] == 0 ? 1 : 0;
                left += 1;
            }
            
            longestOnes = Math.max(longestOnes, right - left + 1);
        }
        
        return longestOnes;
    }
}
